### PR TITLE
Remove deprecated code from Pagerduty provider 

### PR DESCRIPTION
**Pull request by Prab-27**
related: #44559 

remove-deprecated-code, remove test ,update changelog from-pagerduty provider

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/contributing-docs/05_pull_requests.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

File: providers/src/airflow/providers/pagerduty/CHANGELOG.rst [hunk 1]
```diff
@@ -27,6 +27,16 @@
...
 
+main
+.....
+
+.. warning::
+   All deprecated classes, parameters and features have been removed from the pagerduty provider package.
+   The following breaking changes were introduced:
+
+   * Removed deprecated method ``create_event`` from providers.pagerduty.hooks.pagerduty. Please use the ``airflow.providers.pagerduty.hooks.PagerdutyEventsHook`` to interact with the Events API
+   * Removed deprecated method ``create_event`` from providers.pagerduty.hooks.pagerduty_events. Please use the ``PagerdutyEventsHook.send_event`` to interact with the Events API
```
Comment (gopidesupavan) [inline providers/src/airflow/providers/pagerduty/CHANGELOG.rst]:
```suggestion
   * Removed deprecated method ``create_event`` from ``providers.pagerduty.hooks.pagerduty`` and ``providers.pagerduty.hooks.pagerduty_events``. Please use the ``airflow.providers.pagerduty.hooks.pagerduty_events.PagerdutyEventsHook.send_event`` to interact with the Events API
```

Comment (Prab-27) [reply providers/src/airflow/providers/pagerduty/CHANGELOG.rst]: Done

**Review by potiuk — APPROVED** [review]

**Comment by jscheffl** [comment]
Just a static check is failing, when this is fixed I assume this can be merged.

**Comment by Prab-27** [comment]
> Just a static check is failing, when this is fixed I assume this can be merged.

I have modified test_pagerduty.py and another check has failed due to this statement from [providers/src/airflow/providers/pagerduty/hooks/pagerduty.py](url)  
`airflow.providers.pagerduty.hooks.pagerduty_events import PagerdutyEventsHook`

However, fixing the second static check caused the unit test to fail.

```
__ ERROR collecting providers/tests/pagerduty/hooks/test_pagerduty_events.py ___
ImportError while importing test module '/opt/airflow/providers/tests/pagerduty/hooks/test_pagerduty_events.py'.
Hint: make sure your test modules/packages have valid Python names.
```

**Comment by jscheffl** [comment]
> > Just a static check is failing, when this is fixed I assume this can be merged.
> 
> I have modified test_pagerduty.py and another check has failed due to this statement from [providers/src/airflow/providers/pagerduty/hooks/pagerduty.py](url) `airflow.providers.pagerduty.hooks.pagerduty_events import PagerdutyEventsHook`
> 
> However, fixing the second static check caused the unit test to fail.
> 
> ```
> __ ERROR collecting providers/tests/pagerduty/hooks/test_pagerduty_events.py ___
> ImportError while importing test module '/opt/airflow/providers/tests/pagerduty/hooks/test_pagerduty_events.py'.
> Hint: make sure your test modules/packages have valid Python names.
> ```

Yes, because previously the pytest imported "wrong" as a re-import. In `providers/tests/pagerduty/hooks/test_pagerduty_events.py:23` you need to import via:

```
from airflow.providers.pagerduty.hooks.pagerduty_events import PagerdutyEventsHook
```

**Comment by Prab-27** [comment]
Done !

**Comment by jscheffl** [comment]
Green! Merged! Thanks!

**Pull request merged**